pkg/collector/corechecks/cluster/orchestrator/collector_bundle.go

```diff
@@ -0,0 +1,145 @@
+// in order to share/split easily the collector configuration with the KSM core
+// check.
+//
+// If that's not the case then it'll select all available collectors that are
+// marked as stable.
+func NewCollectorBundle(chk *OrchestratorCheck) *CollectorBundle {
+	bundle := &CollectorBundle{
+		check:     chk,
+		inventory: collectors.NewCollectorInventory(),
+		runCfg: &collectors.CollectorRunConfig{
+			APIClient:   chk.apiClient,
+			ClusterID:   chk.clusterID,
+			Config:      chk.orchestratorConfig,
+			MsgGroupRef: &chk.groupID,
+		},
+		stopCh: make(chan struct{}),
+	}
+
+	bundle.prepare()
+
+	return bundle
+}
+
+// prepare initializes the collector bundle internals before it can be used.
+func (cb *CollectorBundle) prepare() {
+	cb.prepareCollectors()
+	cb.prepareExtraSyncTimeout()
+}
+
+// prepareCollectors initializes the bundle collector list.
+func (cb *CollectorBundle) prepareCollectors() {
+	// No collector configured in the check configuration.
+	// Use the list of stable collectors as the default.
+	if len(cb.check.instance.Collectors) == 0 {
+		cb.collectors = cb.inventory.StableCollectors()
+		return
+	}
+
+	// Collectors configured in the check configuration.
+	// Build the custom list of collectors.
+	for _, name := range cb.check.instance.Collectors {
+		if collector, err := cb.inventory.CollectorByName(name); err == nil {
+			if !collector.Metadata().IsStable {
+				_ = cb.check.Warnf("Using unstable collector: %s", name)
+			}
+			cb.collectors = append(cb.collectors, collector)
+		} else {
+			_ = cb.check.Warnf("Unsupported collector: %s", name)
+		}
+	}
+}
+
+// prepareExtraSyncTimeout initializes the bundle extra sync timeout.
+func (cb *CollectorBundle) prepareExtraSyncTimeout() {
+	// No extra timeout set in the check configuration.
+	// Use the default.
+	if cb.check.instance.ExtraSyncTimeoutSeconds <= 0 {
+		cb.extraSyncTimeout = defaultExtraSyncTimeout
+		return
+	}
+
+	// Custom extra timeout.
+	cb.extraSyncTimeout = time.Duration(cb.check.instance.ExtraSyncTimeoutSeconds) * time.Second
+}
+
+// Initialize is used to initialize collectors part of the bundle.
+// During initialization informers are created, started and their cache is
+// synced.
+func (cb *CollectorBundle) Initialize() error {
+	informersToSync := make(map[apiserver.InformerName]cache.SharedInformer)
+
+	for _, collector := range cb.collectors {
+		collector.Init(cb.runCfg)
+		informer := collector.Informer()
+		informersToSync[apiserver.InformerName(collector.Metadata().Name)] = informer
+
+		// we run each enabled informer individually as starting them through the factory
+		// would prevent us to restarting them again if the check is unscheduled/rescheduled
+		// see https://github.com/kubernetes/client-go/blob/3511ef41b1fbe1152ef5cab2c0b950dfd607eea7/informers/factory.go#L64-L66
+		go informer.Run(cb.stopCh)
+	}
+
+	return apiserver.SyncInformers(informersToSync, cb.extraSyncTimeout)
```

Comment:
I know it is not part of this PR but thinking about this I think `WaitForInformersToSync` is more fitting as we actually just wait in this method and don't actually sync there

Reply:
I'm not sure I'm getting this, what do you mean by not syncing informers?

Reply:
Oh what I meant (that is not related to your PR) 
that the function `SyncInformers` doesn't actually `sync` the `informers` but just call `waitForCacheSync` and therefore only wait. 

I think it is more fitting to rename it to `WaitForInformersToSync` (in another PR) 🙈

Reply:
oh ok, yeah since it's used in other places outside the orchestrator check it's probably better to see with stakeholders in another PR